crates/forge_analyzer/src/definitions.rs

```diff
@@ -959,6 +960,25 @@ impl<'cx> FunctionAnalyzer<'cx> {
                 }
             }
         }
+
+        if call_func_with_name(&callee, "then")
+            || call_func_with_name(&callee, "map")
+            || call_func_with_name(&callee, "foreach")
+            || call_func_with_name(&callee, "filter")
+        {
+            if let Some(lambda_function) = args.get(0) {
```

Comment:
Can you add this check to the if statement above? That way, this will handle other stuff like non block statement expressions in arrow functions and function expressions.